lib/googleauth/service_account.rb

```diff
@@ -135,8 +142,13 @@ def self.read_json_key(json_key_io)
       # Initializes a ServiceAccountJwtHeaderCredentials.
       #
       # @param json_key_io [IO] an IO from which the JSON key can be read
-      def initialize(json_key_io)
-        private_key, client_email = self.class.read_json_key(json_key_io)
+      def initialize(json_key_io: nil)
+        if json_key_io
+          private_key, client_email = self.class.read_json_key(json_key_io)
+        else
+          private_key = ENV[CredentialsLoader::PRIVATE_KEY_VAR]
+          client_email = ENV[CredentialsLoader::CLIENT_EMAIL_VAR]
+        end
```

Comment:
This is more readable to me, with less if/else branches. Set default values from ENV first. But whether to change is up to you.
```
  private_key = ENV[CredentialsLoader::PRIVATE_KEY_VAR]
  client_email = ENV[CredentialsLoader::CLIENT_EMAIL_VAR]
  private_key, client_email = self.class.read_json_key(json_key_io) if json_key_io
```